### DISCUSSION TITLE
Need Help with "Maximum number of routes (rewrites, redirects, etc) exceeded." Error deploying Next.js app with next-intl

### DISCUSSION THREAD

**Discussion by PaulKreft** [vercel/next.js#66898] [Help]
### Summary

Has anyone had issues with the "[Routes created per Deployment](https://vercel.com/docs/limits/overview#routes-created-per-deployment)" limit on the hobby plan? I'm using Next.js with next-intl (de + en) which creates 3 routes per page, so I'm quickly exceeding the limit.

I've tried reducing routes through server-side pre-rendering and read that the limit can somehow be bypassed with the use of the Edge Config, which, however, seems to have an absolute usage limit itself (on the hobby plan).

But so far, I've not found a really sustainable solution to create virtually limitless pages. Is that even possible? And why does Vercel have such a limit anyways, makes hosting a blog-like website there kind of pointless, doesn't it?

Has anyone found a good solution to this problem and/or could help me find one? :)
Thanks a lot for your help!

<img width="1267" alt="Screenshot 2024-06-15 at 13 27 33" src="https://github.com/vercel/next.js/assets/64103063/d80121b4-516d-41a6-9fab-7bb74198af5f">

**Comment by joelschneider** [comment]
Interested in this as well.

It's worth noting that you also hit that same limit in the Pro plan.

**Comment by casememos** [comment]
According to new documentation the Maximum number of routes has been increased to 2048.  My deployment was failing because I had 1031 routes created. However, I get the following error now instead of number of routes exceeded:

Unexpected error writing deployment routes

**Comment by gridig** [comment]
> According to new documentation the Maximum number of routes has been increased to 2048.  My deployment was failing because I had 1031 routes created. However, I get the following error now instead of number of routes exceeded:
> 
> 
> 
> Unexpected error writing deployment routes

I also have the "Unexpected error writing deployment routes" error since Friday, I'm not sure it has anything to do with number of routes.

**Comment by marceldekoning** [comment]
I also encounter the same problem. It looks like it has something to do with the amount of redirects we have placed in our projects. In our case, when we add 1200 redirects, we get the error "Unexpected error writing deployment routes". When we add 900 redirects, everything deploys just fine. It looks like there is some kind of redirect cap, but it's not the 2048 that was described here: https://vercel.com/docs/limits/overview.
